baremetal/metal3machine_manager.go

```diff
@@ -149,6 +151,43 @@ func (m *MachineManager) role() string {
 	return bmRoleNode
 }
 
+// SetPauseAnnotation sets the pause annotations on associated bmh
+func (m *MachineManager) SetPauseAnnotation(ctx context.Context) error {
+	// look for associated BMH
+	host, err := m.getHost(ctx)
+	if err != nil {
+		m.setError("Failed to get a BaremetalHost for the Metal3Machine",
+			capierrors.CreateMachineError,
+		)
+		return err
+	}
+	if host == nil {
+		m.Log.Info("BaremetalHost not associated")
+		return nil
+	}
+
+	annotations := host.GetAnnotations()
+
+	if annotations != nil {
+		if _, ok := annotations[bmh.PausedAnnotation]; ok {
+			if host.Annotations[bmh.PausedAnnotation] != pausedAnnotationKey {
+				m.Log.Info("BaremetalHost is already paused by user")
+			} else {
+				m.Log.Info("BaremetalHost is already paused by capM3")
+			}
+			return nil
+		} else {
+			m.Log.Info("Adding PausedAnnotation in BareMetalHost")
+			host.Annotations[bmh.PausedAnnotation] = pausedAnnotationKey
+			return m.updateObject(ctx, host)
```

Comment:
you could take this return and the next one out of the condition, since we return already when the object should not be updated

Reply:
Done